src/main/java/net/glowstone/block/itemtype/ItemBoat.java

```diff
@@ -47,6 +48,7 @@ private void placeBoat(GlowPlayer player) {
             location.setYaw(player.getLocation().getYaw());
             Boat boat = targetBlock.getWorld().spawn(location, Boat.class);
             boat.setWoodType(woodType);
+            InventoryUtil.consumeHeldItem(player, holding);
```

Comment:
Don't you have to forcefully set the slot after changing the amount? IIRC it doesn't update the inventory until you use setSlot().

Reply:
No, that's handled the next time the player is pulsed, by `GlowPlayer` lines 863-5.